File: Completed/python3/1446.py
class Solution:
    def maxPower(self, s: str) -> int:
      if len(s) <= 1:
        return len(s)
      
      tmp = 1
      result = 0
      
      for i in range(1, len(s)):
        prev = s[i-1]
        curr = s[i]
        
        if prev == curr:
          tmp+=1
        else:
          tmp = 1
        
        result = max(result, tmp)
      return result

File: Completed/Go/1446.go
func maxPower(s string) int {
  if len(s) <= 1 {
    return len(s)
  }
  
  tmp, result := 1,0
  for i := 1; i < len(s); i++ {
    prev, curr := s[i-1], s[i]
    if prev == curr {
      tmp++
    } else {
      tmp = 1
    }
    
    if tmp > result {
      result = tmp
    }
  }
  return result
}
